reconcile/test/test_status_page_components.py

```diff
@@ -67,296 +69,275 @@ def f(self):
     return f
 
 
-class TestReconcileLogic(TestCase):
-
-    @staticmethod
-    @patch.object(VaultSecretRef, '_resolve_secret',
-                  new_callable=stub_resolve_secret)
-    @patch.object(AtlassianStatusPage, '_create_component')
-    @patch.object(AtlassianStatusPage, '_fetch_components')
-    def test_create_component(fetch_mock, create_mock, vault_mock):
-        """
-        Test if the creation logic is called for a component missing on the
-        status page.
-        """
-        fixture_name = "test_create_component.yaml"
-        fetch_mock.return_value = \
-            get_atlassian_component_fixtures(fixture_name, "page_1")
-        create_mock.return_value = None
-        page = get_page_fixtures(fixture_name)[0]
-
-        page.reconcile(False, StateStub())
-
-        create_mock.assert_called_with(
-            component_to_dict(page.components[0], "group_id_1")
-        )
+@patch.object(VaultSecretRef, '_resolve_secret',
```

Comment:
I'd suggest moving these to use the `mocker` pytest fixture. The same goes for other tests, but I won't repeat the same comment elsewhere. This is covered more here: https://service.pages.redhat.com/dev-guidelines/docs/appsre/python-testing-guide/#pytest-best-practices

It just helps get away from using decorators because pytest uses those for other uses cases like parameterizing tests, defining fixtures, etc. You'll run into issues of ordering of decorators with the args passed in the test function and it just isn't ideal. The change should be minimal, it'd just look something like:

```python
def test_create_component(mocker):
    fetch_mock = mocker.patch.object(AtlassianStatusPage, '_fetch_components')
    create_mock = ...
    vault_mock = ...

    [the rest of your code stays the same]
    ...
```

Also, while we're fixing these, adding `autospec=True` might not be a bad idea.

Reply:
Thanks for your feedback. I will update accordingly!

Reply:
remove decorators in 1be1e45
use autospec in 2225936